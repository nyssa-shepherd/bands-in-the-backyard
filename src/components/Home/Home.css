.home-page {
    background-color: #fff;
    left: 35%;
    display: flex;
    flex-direction: column;
    align-items: center;
    top: 220px;
    width: fit-content;
}

.cards {
    margin: 0 5%;
}

h2 {
    color: #579197;
    display: block;
    font-size: 30px;
}

h3 {
    color: #579197;
    right: 450px; 
    position: relative;
}

hr {
    right: 490px;
    position: relative;
    width: 1350px;
}

form {
    display: flex;
    align-content: center;
    margin-top: 20px;
    width: 390px;
}

input {
    height: 20px;
    margin-right: 5px;
    width: 85%;
}

input::placeholder {
    font-size: 18px;
}

button {
    background-color: rgba(4, 82, 80, 0.607); 
    border: none;
    color: #fff;
    font-size: 15px;
    height: 25px;
    padding: 5px;
    width: 55px;
}

button:hover {
    background-color: rgba(11, 125, 121, 0.888); 

}