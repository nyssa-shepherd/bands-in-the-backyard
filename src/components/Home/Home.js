import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Search from '../../containers/Search/Search.js';
import CardContainer from '../../containers/CardContainer/CardContainer.js';
import './Home.css';

export class Home extends Component {
  render () {
    let { location } = this.props;
    location = location.toUpperCase();

    return (
      <div className='home-page'>
        <Search />
        <div className='home-header'>
          <h2>UPCOMING NEAR {location}</h2>
        </div>
        <div className='cards'>
          <CardContainer />
        </div>
      </div>
    );
  }
}

export const mapStateToProps = store => ({
  location: store.location
}); 

export default connect(mapStateToProps)(Home);

Home.propTypes = {
  location: PropTypes.string.isRequired
};