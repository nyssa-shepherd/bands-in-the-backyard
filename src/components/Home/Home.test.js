import React from 'react';
import { shallow } from 'enzyme';
import { Home, mapStateToProps } from './Home.js';

describe('Home', () => {
  let wrapper;
  let mockLocation;

  beforeEach(() => {
    mockLocation = 'Denver, CO';
    wrapper = shallow(<Home location={mockLocation}/>);
  });

  it('snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('map the store correctly', () => {
    const mockStore = {location: 'denver'};
    const mapped = mapStateToProps(mockStore);

    expect(mapped.location).toEqual(mockStore.location);
  });
  
});