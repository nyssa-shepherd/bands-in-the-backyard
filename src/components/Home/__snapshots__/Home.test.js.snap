// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    location="Denver, CO"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Connect(Search)) />,
        <div
          className="home-header"
>
          <h2>
                    UPCOMING NEAR 
                    DENVER, CO
          </h2>
</div>,
        <div
          className="cards"
>
          <withRouter(Connect(CardContainer)) />
</div>,
      ],
      "className": "home-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            UPCOMING NEAR 
            DENVER, CO
</h2>,
          "className": "home-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "UPCOMING NEAR ",
              "DENVER, CO",
            ],
          },
          "ref": null,
          "rendered": Array [
            "UPCOMING NEAR ",
            "DENVER, CO",
          ],
          "type": "h2",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <withRouter(Connect(CardContainer)) />,
          "className": "cards",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Connect(Search)) />,
          <div
            className="home-header"
>
            <h2>
                        UPCOMING NEAR 
                        DENVER, CO
            </h2>
</div>,
          <div
            className="cards"
>
            <withRouter(Connect(CardContainer)) />
</div>,
        ],
        "className": "home-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              UPCOMING NEAR 
              DENVER, CO
</h2>,
            "className": "home-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "UPCOMING NEAR ",
                "DENVER, CO",
              ],
            },
            "ref": null,
            "rendered": Array [
              "UPCOMING NEAR ",
              "DENVER, CO",
            ],
            "type": "h2",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <withRouter(Connect(CardContainer)) />,
            "className": "cards",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
